### PR TITLE
Feat/authentification

### PR DESCRIPTION
**Pull request by paulavanina**

### CODE REVIEW

File: src/controllers/fetchBlogsController.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    
+        const sql="SELECT blog_id, titlu, descriere FROM [dbo].[blogs]";
+        const request=new mssql.Request();
+
+        request.query(sql,(err,result)=>{
+            if(err){
+                return res.status(400).json({Error:"eroare in obtinerea datelor"})
+            }
+            const blogsDetails=result.recordset;
+            return res.status(200).json(blogsDetails);
```
Comment (milosanca) [inline src/controllers/fetchBlogsController.js:14]:
```suggestion
            return res.status(200).json(result.recordset);
```

File: src/controllers/loginController.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    const user=result.recordset[0];
+  
+    bcrypt.compare(parola, user.parola, (err, isMatch) => {
+      if (err) {
+        return res.status(500).json({ Error: "Eroare in compararea parolelor." });
+      }
+      if (!isMatch) {
+        return res.status(400).json({ Error: "Credentiale invalide." });
+      }
+      //se genereaza token
```
Comment (milosanca) [inline src/controllers/loginController.js:30]: const {userUUID, email}= user

File: src/controllers/signupController.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import bcrypt from 'bcrypt'
+import dotenv from 'dotenv';
+const signupController=async(req,res)=>{
+    const salt=10;
+    const sql = "INSERT INTO users (userUUID, nume, prenume, email, parola) OUTPUT Inserted.userUuid VALUES (NEWID(), @nume, @prenume, @email, @parola)";
+  const sqlEmail = "SELECT email FROM users WHERE email = @email";
+
+  //verificare email-ului
+  const emailRequest = new mssql.Request();
+  emailRequest.input('email', mssql.VarChar, req.body.email);
```
Comment (milosanca) [inline src/controllers/signupController.js:13]: looks like we  need below a lot of details from req.body maybe destructre here and use it below

File: package.json [hunk 4]
```diff
@@ -2,14 +2,30 @@
...
+    "cors": "^2.8.5",
+    "dotenv": "^16.4.5",
+    "express": "^4.19.2",
+    "express-validator": "^7.2.0",
+    "jsonwebtoken": "^9.0.2",
+    "mssql": "^11.0.1",
+    "react-quill": "^2.0.0",
+    "uuid": "^10.0.0"
+  },
+  "devDependencies": {
```
Comment (milosanca) [inline package.json:28]: what is the difference between `devDependecies` and `dependecies`

File: src/controllers/createBlogController.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import mssql from "mssql";
+import { body, Result } from "express-validator";
+import jwt from "jsonwebtoken";
+
+const createBlogController = async (req, res) => {
+  try {
+    const authHeader = req.headers["authorization"];
+    const token = authHeader && authHeader.split(" ")[1];
```
Comment (milosanca) [inline src/controllers/createBlogController.js:8]: why we slit after empty string?

File: src/controllers/createBlogController.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            .status(400)
+            .json({ Error: "Eroare la inserarea datelor." });
+        }
+        return res
+          .status(200)
+          .json({ Status: "Datele au fost adaugate cu succes in bd." });
+      });
+    });
+  } catch (error) {
+    return res.status(200).json({ Error: "a aparut o eroare" });
```
Comment (milosanca) [inline src/controllers/createBlogController.js:41]: status 200 and we have an error?

File: src/controllers/dbConnectionController.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+      res
+        .status(200)
+        .send({
+          response: dbRes.recordset,
+          status: 200,
+        })
+        .end();
+    })
+    .catch((dbErr) => {
+      console.dbErr("interogarea la baza de date a esuat.");
```
Comment (milosanca) [inline src/controllers/dbConnectionController.js:15]: this line is working?

File: src/controllers/fetchBlogDetailsController.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    request.query(sql, (err, result) => {
+      if (err) {
+        return res.status(400).json({ Error: "eroare in obtinerea datelor" });
+      }
+
+      if (result.recordset.length === 0) {
+        return res.status(404).json({ Error: "utilizatorul nu a fost gasit" });
+      }
+      const blogDetails = result.recordset;
+      return res.status(200).json(blogDetails);
```
Comment (milosanca) [inline src/controllers/fetchBlogDetailsController.js:31]:
```suggestion
      return res.status(200).json(result.recordset);
```

File: src/controllers/fetchBlogsController.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+const fetchBlogsController = async (req, res) => {
+  const sql = "SELECT blog_id, titlu, descriere FROM [dbo].[blogs]";
+  const request = new mssql.Request();
+
+  request.query(sql, (err, result) => {
+    if (err) {
+      return res.status(400).json({ Error: "eroare in obtinerea datelor" });
+    }
+    const blogsDetails = result.recordset;
+    return res.status(200).json(blogsDetails);
```
Comment (milosanca) [inline src/controllers/fetchBlogsController.js:13]:
```suggestion
    return res.status(200).json(result.recordset);
```

File: src/routes/fetchBlogsRoute.js [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import { Router } from "express";
+import fetchBlogsController from "../controllers/fetchBlogsController.js";
+const fetchBlogsRoute = Router();
+
+fetchBlogsRoute.get("/blogs", (req, res) => {
+  return fetchBlogsController(req, res);
+});
```
Comment (milosanca) [inline src/routes/fetchBlogsRoute.js:7]:
```suggestion
fetchBlogsRoute.get("/blogs", (req, res) => 
  fetchBlogsController(req, res);
);
```

File: src/routes/userProfileRoute.js [hunk 11]
```diff
@@ -0,0 +1,10 @@
+import { Router } from "express";
+import userProfileController from "../controllers/userProfileController.js";
+
+const userProfileRoute = Router();
+
+userProfileRoute.get("/user-profile", (req, res) => {
+  return userProfileController(req, res);
+});
```
Comment (milosanca) [inline src/routes/userProfileRoute.js:8]: same as above

File: src/routes/signupRoute.js [hunk 12]
```diff
@@ -0,0 +1,10 @@
+import { Router } from "express";
+import signupController from "../controllers/signupController.js";
+
+const signupRoute = Router();
+
+signupRoute.post("/signup", (req, res) => {
+  return signupController(req, res);
+});
```
Comment (milosanca) [inline src/routes/signupRoute.js:8]: same as above

File: src/routes/loginRoute.js [hunk 13]
```diff
@@ -0,0 +1,10 @@
+import { Router } from "express";
+import loginController from "../controllers/loginController.js";
+
+const loginRoute = Router();
+
+loginRoute.post("/login", (req, res) => {
+  return loginController(req, res);
+});
```
Comment (milosanca) [inline src/routes/loginRoute.js:8]: same as above

File: src/controllers/createBlogController.js (hunk 5 above)
Comment (paulavanina) [reply src/controllers/createBlogController.js:8]: const config = {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        }; this is how i configured token in frontend and i used split to return the second element which is the token

File: src/controllers/dbConnectionController.js (hunk 7 above)
Comment (paulavanina) [reply src/controllers/dbConnectionController.js:15]: yes

File: src/controllers/createBlogController.js (hunk 5 above)
Comment (milosanca) [reply src/controllers/createBlogController.js:8]: alright maybe split after 'Bearer'?

**Pull request merged**